baseplate/sidecars/trace_publisher.py

```diff
@@ -179,10 +190,17 @@ def publish_traces() -> None:
     while True:
         message: Optional[bytes]
 
-        try:
-            message = trace_queue.get(timeout=0.2)
-        except TimedOutError:
-            message = None
+        if publisher_cfg.queue_type == "in_memory":
+            with publisher_queue_utils.start_queue_server(host="127.0.0.1", port=9090):
```

Comment:
This will start a queue server and then kill it for every individual message. Is that what you want? Would it not be better to start the server once outside the while loop, then have the while loop be based on a boolean linked to if any errors have been encountered? If they are then the loop can end and the context manager can clean up the queue.